src/ng/directive/input.js

```diff
@@ -1126,7 +1128,11 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     });
   }
 
-  var timeout;
+  var timeout, oldVal;
+  var viewValueUpdated = true, msieInput = msie >= 10 && msie <= 11;
+  if (msieInput && attr.type in IE_INPUTS_WITH_CLEARING) {
```

Comment:
Why do we even care about the type for setting the initial `oldVal` though? I'd think we'd always want to setup the `oldVal` if `msieInput` is true...